scipy/sparse/linalg/_isolve/tests/test_gcrotmk.py

```diff
@@ -66,18 +66,18 @@ def test_preconditioner(self):
     def test_arnoldi(self):
         np.random.seed(1)
 
-        A = eye(2000) + rand(2000, 2000, density=5e-4)
+        A = eye_array(2000) + random_array((2000, 2000), density=5e-4)
         b = np.random.rand(2000)
 
         # The inner arnoldi should be equivalent to gmres
         with suppress_warnings() as sup:
             sup.filter(DeprecationWarning, ".*called without specifying.*")
-            x0, flag0 = gcrotmk(A, b, x0=zeros(A.shape[0]), m=15, k=0, maxiter=1)
-            x1, flag1 = gmres(A, b, x0=zeros(A.shape[0]), restart=15, maxiter=1)
+            x0, flag0 = gcrotmk(A, b, x0=zeros(A.shape[0]), m=10, k=0, maxiter=1)
+            x1, flag1 = gmres(A, b, x0=zeros(A.shape[0]), restart=10, maxiter=1)
```

Comment:
Why are these parameters changing?

Reply:
Thanks for this question. I should have discussed it in the OP.

Changing from `rand` to `random_array` changes the specific random numbers generated for a specific seed. This test sets the seed to `1` for reproducability, but it fails for other seeds. Two types of failures occur. For some seeds, the `flag1` output from `gmres` indicates it did not converge. For others, the norm residuals are not large enough to pass the lower-bound-on-residual set to 1e-3 in line 80 below. It turns out that with `m=15` convergence was actually rare as the seed is varied. The choice of seed `1` was fortuitous. Switching the `gmres` parameter to `m=10` made that better and so I had to switch the parameter value for `gcrotmk` to match. With those params, the lower-bound-on-residual cutoff was robust to changing `seed` if set to 1e-4 (.

So, this test has changed -- but it was fragile before. I believe this is a better test.

Some day we will switch from `RandomState` to `default_rng`/`Generators` and we will run into this again. Hopefully the new parameter and residual level with be robust enough for that. Should I switch to `Generator` here now?